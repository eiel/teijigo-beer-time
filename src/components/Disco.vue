<template>
  <div class="disco">
    <h1>DISCOGRAPHY</h1>
    <p>★★★★★★★工事中★★★★★★★</p>
    <img src="../assets/job_kouji_ojigi.png" alt="工事作業員のイラスト「おじぎ・お詫び」">
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';

@Component
export default class Disco extends Vue {}
</script>

<style scoped>
</style>
