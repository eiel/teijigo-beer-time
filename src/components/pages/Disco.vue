<template>
  <default-template>
    <div class="disco">
      <tbt-h1>DISCOGRAPHY</tbt-h1>
      <p>★★★★★★★工事中★★★★★★★</p>
      <img src="@/assets/job_kouji_ojigi.png" alt="工事作業員のイラスト「おじぎ・お詫び」">
    </div>
  </default-template>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import DefaultTemplate from '@/components/templates/Default';
import TbtH1 from '@/components/atoms/TbtH1';

@Component({
  components: {
    DefaultTemplate,
    TbtH1,
  },
})
export default class Disco extends Vue {}
</script>

<style scoped>
</style>
