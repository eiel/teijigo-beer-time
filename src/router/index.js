import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top';
import Disco from '@/components/Disco';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
  ],
});
