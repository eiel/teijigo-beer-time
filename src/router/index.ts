import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top.vue';
import Disco from '@/components/Disco.vue';
import NotFound from '@/components/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
});
