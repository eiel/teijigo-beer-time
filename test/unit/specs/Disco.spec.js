import Vue from 'vue';
import Disco from '@/components/Disco';
import router from '@/router/index.ts';

describe('Disco.vue', () => {
  it('should render correct contents', () => {
    const Constructor = Vue.extend(Disco);
    const vm = new Constructor({ router }).$mount();
    expect(vm.$el.querySelector('.disco h1').textContent)
      .toEqual('DISCOGRAPHY');
  });
});
