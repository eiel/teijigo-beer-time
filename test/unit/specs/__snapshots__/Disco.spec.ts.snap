// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disco.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-active"
          href="/"
        >
          
        トップ
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
          
        DISCOGRAPHY
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
          
        メンバー紹介
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
          
        一言掲示板
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/beenya"
        >
          
        びーにゃの部屋
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="disco"
    >
      <h1
        class="h1"
      >
        DISCOGRAPHY
      </h1>
       
      <p>
        ★★★★★★★工事中★★★★★★★
      </p>
       
      <img
        alt="工事作業員のイラスト「おじぎ・お詫び」"
        src="@/assets/job_kouji_ojigi.png"
      />
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
